src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -586,6 +600,38 @@ public String call(I item) {
         }
     }
 
+    private static void onExtendTimeout(long timeout, @NonNull TimeUnit unit) {
+        // TODO When baseline is updated to jenkinsci/jenkins#6237, use Jenkins.get().getLifecycle().onExtendTimeout(timeout, unit)
+        Method onExtendTimeout;
+        try {
+            onExtendTimeout = Lifecycle.class.getMethod("onExtendTimeout", long.class, TimeUnit.class);
+        } catch (NoSuchMethodException e) {
+            return;
+        }
+
+        Lifecycle lifecycle = Jenkins.get().getLifecycle();
+        try {
+            onExtendTimeout.invoke(lifecycle, timeout, unit);
+        } catch (IllegalAccessException e) {
+            IllegalAccessError x = new IllegalAccessError(e.getMessage());
+            x.initCause(e);
+            throw x;
+        } catch (InvocationTargetException e) {
+            Throwable t = e.getCause();
+            if (t instanceof RuntimeException) {
+                throw (RuntimeException) t;
+            } else if (t instanceof IOException) {
```

Comment:
is not documented to be thrown in https://github.com/jenkinsci/jenkins/blob/0c4b79128d1c437955f62d8c4332301f28bb8615/core/src/main/java/hudson/lifecycle/Lifecycle.java#L309 how can this happen?